File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema24Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test


class Schema24Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine;

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema08Test::class.java.getResourceAsStream("/schema24.dbs"))
    }

   @Test
   fun `the usual stuff`() {
       val reference = mapOf<String,Any>("ner" to arrayOf(mapOf("text" to "asdf1234")))
       val serialized = engine.serialize(reference)
       val deserialized = engine.deserialize(serialized)
       Assertions.assertThat(deserialized["ner"]).isEqualTo(reference["ner"])
   }
}

File: dynabuffers-python/tests/usecase/Schema24Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema24(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema24.dbs"))

    def test(self):
        obj = {"ner": [{"text": "asdf1234"}]}
        result = self.engine.deserialize(self.engine.serialize(obj, ["outgoing"]))
        self.assertEqual(result, obj)


if __name__ == "__main__":
    unittest.main()
